l3build-arguments.lua

```diff
@@ -298,22 +292,20 @@ options = argparse()
 
 -- Sanity check
 function check_engines(config)
-  if options["engine"] and not options["force"] then
+  if options["engine"] then
     -- Make a lookup table
     local t = { }
     for _, engine in pairs(checkengines) do
       t[engine] = true
     end
-    for _, engine in pairs(options["engine"]) do
-      if not t[engine] then
-        print("\n! Error: Engine \"" .. engine .. "\" not set up for testing with configuration \"" .. config .. "\"!")
-        print("\n  Valid values are:")
-        for _, engine in ipairs(checkengines) do
-          print("  - " .. engine)
-        end
-        print("")
-        exit(1)
+    checkengines = {}
+    for _,engine in ipairs(options["engine"]) do
+      if t[engine] then
+          insert(checkengines,engine)
+      else
+        print("Skipping unknown engine " .. engine)
       end
     end
   end
+  if not next(checkengines) then exit(0) end
```

Comment:
not sure it helps or not, but perhaps add a print("No applicable engine requested, config ignored")